lib/ansible/modules/cloud/cloudstack/cs_nic.py

```diff
@@ -205,14 +214,14 @@ def get_secondary_ip(self):
 
     def present_nic(self):
         nic = self.get_nic()
-        if not self.get_secondary_ip():
+        if self.vm_guest_ip and not self.get_secondary_ip():
```

Comment:
Ah, now I understand. Your intention is to have the functionality to add a nic *without* requesting an IP to an existing VM? In this case let us add another param, yes. `add_ip` would be fine.